frontend/app/authSlice.ts

```diff
@@ -1,69 +1,65 @@
-import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import {createAsyncThunk, createSlice, PayloadAction} from '@reduxjs/toolkit';
 import * as Keychain from 'react-native-keychain';
 
-export type AuthStatus = 'loading' | 'authenticated' | 'unauthenticated';
+/**
+ * Auth status can be one of the following:
+ * - 'loading': The initial status, before we've loaded the token
+ * - 'authenticated': The user is authenticated and we have a valid token
+ * - 'unauthenticated': There is no token to load and the user is not authenticated
+ * - 'stale': The access token has expired and we are refreshing it
+ */
+export type AuthStatus =
+  | 'loading'
+  | 'authenticated'
+  | 'unauthenticated'
+  | 'stale';
 
 interface AuthState {
-  token: string;
+  refreshToken: string;
+  accessToken: string;
   status: AuthStatus;
-  isAuthenticated: boolean;
 }
 
 const initialState: AuthState = {
-  token: '',
-  status: 'loading', 
-  isAuthenticated: false,
+  refreshToken: '',
+  accessToken: '',
+  status: 'loading',
 };
 
-
-export const logout = createAsyncThunk(
-    'auth/logout',
-    Keychain.resetInternetCredentials('greengrub')
-    // Have this part for later when we have access and refresh
-    // async (_, {dispatch, getState}) => {
-    //   try {
-    //     const result = await fetch('http://localhost:8000/api/logout/', {
-    //       headers: {
-    //         Authorization: `Token ${
-    //           (getState() as {auth: {token: string}}).auth.token
-    //         }`,
-    //       },
-    //     });
-    //     if (result.ok) {
-    //       await Keychain.resetInternetCredentials('greengrub');
-    //     } else {
-    //       console.log('Could not logout');
-    //     }
-    //   } catch (error) {
-    //     console.log(error);
-    //   }
-    // },
-  );
+export const logout = createAsyncThunk('auth/logout', async () => {
+  Keychain.resetInternetCredentials('greengrub');
+});
 
 export const loadAuthToken = createAsyncThunk('auth/loadToken', async () => {
   let status: AuthStatus = 'loading';
-  let token: string = '';
+  let refreshToken: string = '';
   try {
     const credentials = await Keychain.getInternetCredentials('greengrub');
     if (credentials) {
-      token = credentials.password; 
-      status = 'authenticated';
+      refreshToken = credentials.password;
+      status = 'stale';
```

Comment:
Don't know what stale is but will read on it, otherwise makes sense

Reply:
Okay, will add docs to it. It is a state when the access token has expired. I didn't really use it, but it's there if we ever want to use it.